pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -652,6 +654,12 @@ func applyConnectionPool(env *model.Environment, cluster *apiv2.Cluster, setting
 		if settings.Http.Http2MaxRequests > 0 {
 			// Envoy only applies MaxRequests in HTTP/2 clusters
 			threshold.MaxRequests = &types.UInt32Value{Value: uint32(settings.Http.Http2MaxRequests)}
+			if pilot.EnableH2AutoUpgrade() {
+				// by configuring http2 settings for a destination
```

Comment:
I don't believe that's true. People copy&paste settings and they can have both HTTP/1.1 and HTTP/2 options configured, regardless of the supported protocol.

Also, we're supposed to upgrade only in-mesh traffic, at least for the time being.

Reply:
Why would you have http2 options for an http1 destination ?
I understand that envoy api uses the option if appropriate, but for the istio api, why would someone set h2 option randomly ?